### DISCUSSION TITLE
Welcome to AI SDK for SAP Discussions!

### DISCUSSION THREAD

**Discussion by gopalnair** [microsoft/aisdkforsapabap#3] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by skt1483** [comment]
Thanks Gopal for sharing a fantastic way to easily integrate.

Do we have any package/sdk to connect Generative AI ( Like ChatGPT) with SAP?

**Comment by gopalnair** [comment]
Hi Saurabh,
    The current version (v 1.0) of AI SDK, supports GPT 3. However, soon - as soon as we have completed the documentation of current features in documentation site (it's almost done)- the very next thing - is the generative AI support - GPT 3.5 turbo and GPT 4. It's coming very soon!

Best Regards,
Gopal Nair.

**Comment by vj0924** [comment]
Hello, I'm trying to install and use the AI SDK. Thank you for providing this SDK.

One question is how AI SDK for SAP interfaces with Azure OpenAI Service is REST? I'm asking you a question because I'm an ABAP developer and I don't know the network well.

**Reply by gopalnair** [reply]
Hi,
 Yes... under the hood, there is REST api that interacts with the AI engine - be in Azure Open AI, or Open AI. However, before it reaches that stage, there are a number of processing and control points.

Best Regards,
Gopal Nair.

**Comment by Roypalatty** [comment]
Hi,
I would like to know whether this model is additionally trained to generate SAP abap code? 
Example :  If I give a business logic from order management , would it pick tables from SAP and generate a basic code/ function? or it is just do the same as Open AI ( chat GPT) ?
Thanks,
Roy P
SAP Technical Architect

**Reply by gopalnair** [reply]
Hi @Roypalatty ,
   GPT-4 models seems to be more proficient in ABAP programming language than Davinci 003. Infact, the demo program for Chat Completions (which is GPT-4 models) - has sample code which generates ABAP code.

Here is the [Documentation website page for Chat Completions](https://microsoft.github.io/aisdkforsapabap/docs/working-with-sdk/chat_completions) - which shows the demo program name for Azure Open AI or Open AI. Please do take a look at the code. The emphasis on that program is to act as a learning resource - so, have tried to put a lot of comments - almost in every step.


Please let me know how it goes, and if there are additional information/improvements to be made in the documentation/program, we can certainly discuss and make that happen.

Best Regards,
Gopal Nair.

**Reply by Roypalatty** [reply]
Sure. Thank you.

**Comment by raghav04101987** [comment]
Hi Gopal,

We are trying to evaluate the proposition of having the Microsoft AI SDK for ABAP development in SAP GUI 8.0 version and non HANA platform. We would like to know the security feature of the SDK. The question may seem trivial but still asking that what is the code security options and non disclosure of the code to the open source environment after LLM engine goes through the code repository of the client.
